src/View/AjaxView.php

```diff
@@ -33,23 +29,11 @@ class AjaxView extends View
     protected $layout = 'ajax';
 
     /**
-     * Constructor
-     *
-     * @param \Cake\Http\ServerRequest|null $request The request object.
-     * @param \Cake\Http\Response|null $response The response object.
-     * @param \Cake\Event\EventManager|null $eventManager Event manager object.
-     * @param array $viewOptions View options.
+     * @inheritDoc
      */
-    public function __construct(
-        ?ServerRequest $request = null,
-        ?Response $response = null,
-        ?EventManager $eventManager = null,
-        array $viewOptions = []
-    ) {
-        if ($response) {
-            $response = $response->withType('ajax');
-        }
-
-        parent::__construct($request, $response, $eventManager, $viewOptions);
+    public function initialize(): void
+    {
+        parent::initialize();
+        $this->setResponse($this->getResponse()->withType('ajax'));
```

Comment:
There's a tiny chance this can cause problem for someone who has overridden `initialize()` and forgot to call `parent::initialize()` (as you did in previous commit :slightly_smiling_face: ). So it would be safer to target this to `4.next`.

Reply:
Sounds good.